File: killrvideo-service-comments/src/main/java/com/killrvideo/service/comment/dto/CommentByUser.java
package com.killrvideo.service.comment.dto;

import java.util.UUID;

import com.datastax.oss.driver.api.mapper.annotations.CqlName;
import com.datastax.oss.driver.api.mapper.annotations.Entity;
import com.datastax.oss.driver.api.mapper.annotations.PartitionKey;

/**
 * Specialization for USER.
 *
 * @author DataStax Developer Advocates team.
 */
@Entity
@CqlName("comments_by_user")
public class CommentByUser extends Comment {
    
    /** Serial. */
    private static final long serialVersionUID = 1453554109222565840L;
    
    /**
     * Default constructor.
     */
    public CommentByUser() {}
    
    /**
     * Copy constructor.
     *
     * @param c
     */
    public CommentByUser(Comment c) {
        this.commentid  = c.getCommentid();
        this.userid     = c.getUserid();
        this.videoid    = c.getVideoid();
        this.comment    = c.getComment();
    }

    /**
     * Getter for attribute 'userid'.
     *
     * @return
     *       current value of 'userid'
     */
    @PartitionKey
    public UUID getUserid() {
        return userid;
    }

    public Comment toComment() {
        Comment comment = new Comment();
        comment.setCommentid(this.commentid);
        comment.setUserid(this.userid);
        comment.setVideoid(this.videoid);
        comment.setComment(this.comment);
        return comment;
    }
}

File: killrvideo-service-comments/src/main/kotlin/com/killrvideo/service/comment/dto/CommentByUser.kt
package com.killrvideo.service.comment.dto

import com.datastax.oss.driver.api.mapper.annotations.*
import java.time.Instant
import java.util.*
import javax.validation.constraints.NotNull
import javax.validation.constraints.Size

/**
 * Specialization for USER.
 *
 * @author DataStax Developer Advocates team.
 */
@Entity
@CqlName("comments_by_user")
data class CommentByUser(
    @PartitionKey
    var userid: @NotNull UUID?,
    @ClusteringColumn
    var commentid: @NotNull UUID?,
    var comment: @Size(min = 1, message = "The comment must not be empty") String?,
    var videoid: @NotNull UUID?,
    @Computed("toTimestamp(commentid)")
    var dateOfComment: @NotNull Instant?
) {
    fun toComment(): Comment =
        Comment(
            userid = this.userid,
            videoid = this.videoid,
            commentid = this.commentid,
            comment = this.comment,
            dateOfComment = this.dateOfComment
        )

    companion object {
        fun from(comment: Comment): CommentByUser =
            CommentByUser(
                userid = comment.userid,
                videoid = comment.videoid,
                commentid = comment.commentid,
                comment = comment.comment,
                dateOfComment = comment.dateOfComment
            )
    }
}